### ISSUE TITLE
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain" 

### ISSUE THREAD

**Issue by liudonghua123** [jupyterlite/jupyterlite#569]
## Description

I tried to execute `jupyter lite init` and `jupyter lite serve`, then I opened `http://127.0.0.1:8000/index.html` in the latest chrome 99 browser. The page is blank and I check the console log and found `ailed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.` around `config-utils.js:1 `.

**_If I use `serve` of npm to serve the static files. No such problems._**

So it's maybe a simple python web server MIME type config error. I will try to check the code and fix it.

![image](https://user-images.githubusercontent.com/2276718/159116542-e9c56926-af06-4cac-87e0-69fee3fbfc91.png)

## Reproduce

1. `jupyter lite init`
2. `jupyter lite serve`
3. Open `http://127.0.0.1:8000/index.html` in the latest chrome 99 browser

## Context

- JupyterLite version: 0.1.0b3
- Operating System and version: Microsoft Windows [Version 10.0.22000.556]
- Browser and version: Chrome 99

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

**Comment by liudonghua123** [comment]
I find the actural serve used is `tornado`, and I also tried `python -m http.server --directory _output 5000`, the error is the same. So both `tornado` and `http.server` should config the mime type mappings directly.

The code need to update maybe the following.

https://github.com/jupyterlite/jupyterlite/blob/5889b98f0f3df0317d87a518a3e9294c9f0edf0a/py/jupyterlite/src/jupyterlite/addons/serve.py#L57-L68

```python
# a demonstation code from https://gist.github.com/HaiyangXu/ec88cbdce3cdbac7b8d5?permalink_comment_id=3275090#gistcomment-3275090
import http.server
import socketserver

PORT = 8080

class HttpRequestHandler(http.server.SimpleHTTPRequestHandler):
    extensions_map = {
        '': 'application/octet-stream',
        '.manifest': 'text/cache-manifest',
        '.html': 'text/html',
        '.png': 'image/png',
        '.jpg': 'image/jpg',
        '.svg':	'image/svg+xml',
        '.css':	'text/css',
        '.js':'application/x-javascript',
        '.wasm': 'application/wasm',
        '.json': 'application/json',
        '.xml': 'application/xml',
    }

httpd = socketserver.TCPServer(("localhost", PORT), HttpRequestHandler)

try:
    print(f"serving at http://localhost:{PORT}")
    httpd.serve_forever()
except KeyboardInterrupt:
    pass
```


For `tornado`, `get_content_type` in `tornado.web.StaticFileHandler` implementation (https://github.com/tornadoweb/tornado/blob/3de78b7a15ba7134917a18b0755ea24d7f8fde94/tornado/web.py#L2890-L2909) used `mimetypes.guess_type` which is not always correct for js files. Need to overwrite to force set mime type of js files.

```shell
C:\Users\Liu.D.H>python
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import mimetypes
>>> mimetypes.guess_type(r"D:\code\python\jupyterlite\test\_output\config-utils.js")
('text/plain', None)
>>> mimetypes.guess_type(r"D:\code\python\jupyterlite\test\_output\bootstrap.js")
('text/plain', None)
>>>
```

https://github.com/jupyterlite/jupyterlite/blob/5889b98f0f3df0317d87a518a3e9294c9f0edf0a/py/jupyterlite/src/jupyterlite/addons/serve.py#L90-L100

```python
    def _serve_tornado(self):
        from tornado import ioloop, web

        class Handler(web.StaticFileHandler):
            # ...other code
            def get_content_type(self):
                if self.absolute_path.endswith('.js'):
                    return 'application/x-javascript'
                return super().get_content_type()
```

**Comment by bollwyvl** [comment]
Given that [python 3.10](https://github.com/python/cpython/blob/v3.10.3/Lib/mimetypes.py#L430) ships the correct information, it's possible your [system's registry configuration](https://bugs.python.org/issue32462) may be at issue. We have, gingerly, added some of these upstream on `jupyter_server` and `notebook`, and it's pretty much a losing proposition to get them all right if the underlying system is misconfigured.

**Comment by liudonghua123** [comment]
@bollwyvl Thanks for your tips. However I tested on a few other windows computers. It's the same problem. It seems to be a windows platform specific problem.

**Comment by bollwyvl** [comment]
Did you look at the underlying registry key, as documented [here](https://bugs.python.org/issue43975) or [here](https://bugs.python.org/issue10551)? 

`HKEY_CLASSES_ROOT\.js\Content Type was 'text/plain'`

**Comment by liudonghua123** [comment]
> Did you look at the underlying registry key, as documented [here](https://bugs.python.org/issue43975) or [here](https://bugs.python.org/issue10551)? 
> 
> `HKEY_CLASSES_ROOT\.js\Content Type was 'text/plain'`

Hi, I verified "HKEY_CLASSES_ROOT\.js\Content Type" on a few computers, all of them is 'text/plain', it seems like the default value.

**Comment by bollwyvl** [comment]
Some of this was addressed in #670 : please open a new issue if something else remains.
